lms/djangoapps/oauth_dispatch/tests/test_views.py

```diff
@@ -0,0 +1,142 @@
+"""
+Tests for Blocks Views
+"""
+
+import json
+
+import ddt
+from django.test import TestCase, RequestFactory
+from unittest import skip
+
+from student.tests.factories import UserFactory
+
+from .. import adapters
+from .. import views
+
+
+@ddt.ddt
+class TestAccessTokenView(TestCase):
+    """
+    Test class for AccessTokenView
+    """
+
+    dop_adapter = adapters.DOPAdapter()
+    dot_adapter = adapters.DOTAdapter()
+
+    def setUp(self):
+        super(TestAccessTokenView, self).setUp()
+        self.user = UserFactory()
+        self.dot_app = self.dot_adapter.create_public_client(user=self.user, client_id='dot-app-client-id')
+        self.dop_client = self.dop_adapter.create_public_client(user=self.user, client_id='dop-app-client-id')
+
+    def _post_access_token_request(self, user, client):
+        """
+        Make an HTTP POST request a token for the given user and client.
+
+        Returns an HTTP Response.
+        """
+        token_view = views.AccessTokenView.as_view()
+        request = RequestFactory().post('/', {
+            'client_id': client.client_id,
+            'grant_type': 'password',
+            'username': user.username,
+            'password': 'test',
+        })
+        return token_view(request)
```

Comment:
It's faster, and keeps the test a unit test (rather than running all of the middleware every time you want to run your view code). That's a big part of why the request factory was created.  I'm actually going to switch it to the other way, though, because some of the views I'm testing require session data, and I don't want to take the time to figure out how to populate the request object properly.